### DISCUSSION TITLE
Is there a way to have a persistent bash environment across all chunks?

### DISCUSSION THREAD

**Discussion by jmrussell** [quarto-dev/quarto-cli#10082] [Q&A]
### Description

```{bash}
echo "Hello, world!"
foo=bar
```
Some text here describing something

```{bash}
echo $foo
```

I'm curious if there is a way to get behavior like this to work. AFAIK the second code chunk prints an empty line w/ both knitr and jupyter engines because it's executing bash independently in each block

**Comment by mcanouil** [comment]
If following the documentation does not work, then no it's not possible (https://quarto.org/docs/computations/execution-options.html#shell-commands).
Note that Quarto only uses the engines (jupyter, julia, knitr), it has no control on how they work.
